src/base/dttvar.f90

```diff
@@ -939,6 +942,108 @@ subroutine dttvar &
 
   endif
 
+!===============================================================================
+! 4.6 PRINT COMBINED COURANT/FOURIER NUMBER FOR TRANSPORTED SCALARS
+!===============================================================================
+
+  do ii = 1, nscal
+
+    ! Get field parameters
+    call field_get_key_struct_var_cal_opt(ivarfl(isca(ii)), vcopt_t)
+
+    if ( vcopt_t%idiff.ge.1 .or. vcopt_t%iconv.ge.1 ) then
+
+      ! Get mass fluxes
+      call field_get_key_int(ivarfl(isca(ii)), kimasf, iflmas)
+      call field_get_key_int(ivarfl(isca(ii)), kbmasf, iflmab)
+      call field_get_val_s(iflmas, imasflt)
+      call field_get_val_s(iflmab, bmasflt)
+
+      ! Get density
+      call field_get_key_int(ivarfl(isca(ii)), kromsl, flid)
+      if (flid.gt.-1) then
+        call field_get_val_s(flid, cromt)
+      else
+        call field_get_val_s(icrom, cromt)
+      endif
+
+      ! Compute diffusivity
+      if (vcopt_t%idiff.ge.1) then
+
+        call field_get_key_double(ivarfl(isca(ii)), ksigmas, prt)
+        call field_get_key_int(ivarfl(isca(ii)), kivisl, flid)
+        if (flid.gt.-1) then
+          call field_get_val_s(flid, viscl)
+          do iel = 1, ncel
+            w1(iel) = viscl(iel) + vcopt_t%idifft*visct(iel)/prt
+          enddo
+        else
+          do iel = 1, ncel
+            w1(iel) = visls0(ii) + vcopt_t%idifft*visct(iel)/prt
+          enddo
+        endif
+        call viscfa(imvisf, w1, viscf, viscb)
+
+      else
+
+        do ifac = 1, nfac
+          viscf(ifac) = 0.d0
+        enddo
+        do ifac = 1, nfabor
+          viscb(ifac) = 0.d0
+        enddo
+
+      endif
+
+      ! Boundary conditions for matrdt
+      do ifac = 1, nfabor
+        if (bmasflt(ifac).lt.0.d0) then
+
+          iel = ifabor(ifac)
+          if (flid.gt.-1) then
```

Comment:
Here, maybe we should make sure flid points to the field kivisl. It is not the case when idiff=0

Reply:
Fixed by new commit.